persist/fs/commitlog/reader.go

```diff
@@ -109,58 +165,171 @@ func (r *reader) Read() (
 	annotation ts.Annotation,
 	resultErr error,
 ) {
-	entry, err := r.readEntry()
-	if err != nil {
-		resultErr = err
-		return
+	if r.nextIndex == 0 {
+		go r.readLoop()
```

Comment:
afaict, the code assumes there should only be a single `readLoop` running at any point (`readChunk` is appending data to the bytes array without a guard). Could you add a lock/guard here or in `readLoop` to ensure that only a single instance runs at a time

Reply:
Sure, what are you thinking? I guess I could do something like when readLoop starts it locks a global counter, makes sure that the counter is equal to zero and if its not...panic?

Reply:
yeah something like a bool w/ a guard or an atomic value in the reader struct. Lol, no panics, how bout just an error?

Reply:
Sure, should I just return the error from the Read() call? I wanted to add the check in the readLoop() call itself, but I can't return errors from that because I'm executing it async with go keyword.

I could either:

1) Do the check inside of the Read() function (protects callers from misuse, but not people writing code in this file)
2) do the check in the readLoop function, remove the go keyword in the caller, and make readLoop() start its own goroutine after validating that its safe to do so.

Do you have a preference?

Reply:
Either way is good with me.